pkgs/development/libraries/libxml2/default.nix

```diff
@@ -46,6 +46,11 @@ libxml = stdenv.mkDerivation rec {
     hash = "sha256-jI8QkjQKif8yvEStXJaTr/m8ino+FhuyOWZuXRWsmqo=";
   };
 
+  patches = [
+    # Backport that fixes perlPackages.XMLLibXML
+    ./io-Fix-memory-lifetime-issue-with-input-buffers.patch
```

Comment:
We might want to fetch it from the 2.12 branch `https://gitlab.gnome.org/GNOME/libxml2/-/commit/1b6362ea443192a5334b9e682e4725dcd1e2071e.patch`

Or event just point to [2.12 branch HEAD](https://gitlab.gnome.org/GNOME/libxml2/-/commits/2.12/?ref_type=heads) since 2.12.4 is [planned](https://github.com/shlomif/perl-XML-LibXML/issues/81#issuecomment-1854165670) but not here yet.

Reply:
Looking at the patches, we do not care about the `--without-output` but https://gitlab.gnome.org/GNOME/libxml2/-/commit/d7714f977d1670c475d45600fb07668b2505605e might be fixing some stuff (though there does not appear to be any issue referenced).

Reply:
> Or event just point to [2.12 branch HEAD](https://gitlab.gnome.org/GNOME/libxml2/-/commits/2.12/?ref_type=heads) since 2.12.4 is [planned](https://github.com/shlomif/perl-XML-LibXML/issues/81#issuecomment-1854165670) but not here yet.

Done.